include/ginkgo/core/distributed/preconditioner/schwarz.hpp

```diff
@@ -150,6 +169,26 @@ class Schwarz
     void set_solver(std::shared_ptr<const LinOp> new_solver);
 
     std::shared_ptr<const LinOp> local_solver_;
+
+    /**
+     * Manages a vector as a cache, so there is no need to allocate one every
+     * time an intermediate vector is required.
+     * Copying an instance will only yield an empty object since copying the
+     * cached vector would not make sense.
+     *
+     * @internal  The struct is present so the whole class can be copyable
+     *            (could also be done with writing `operator=` and copy
+     *            constructor of the enclosing class by hand)
+     */
+    mutable struct cache_struct {
```

Comment:
I think this should be a `DenseCache` instead.

Reply:
I think it does not support distributed::Vector.

Reply:
you're right. Perhaps it would make sense to add `VectorCache`, instead of using the cache construct.